src/tests/test_authorization_interactions.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+from requests.exceptions import HTTPError, JSONDecodeError, Timeout
+
+from types import SimpleNamespace
+
+from datawave_cli.authorizations_interactions import AuthorizationInteractions, main
+from tests.utils import create_mock_requests_get
+
+
+@pytest.fixture
+def mock_args():
+    args = SimpleNamespace(
+        url="https://test.com",
+        ip=False,
+        localhost=False,
+        cert="fake_cert",
+        key="fake_key",
+        header=[("Authorization", "Bearer fake_token")],
+        namespace="test_namespace",
+        log_level="INFO",
+        view=False
+    )
+    return args
+
+
+@pytest.fixture
+def authorization_interactions(mock_args, mocker):
+    mock_logger = mocker.Mock()
+    return AuthorizationInteractions(mock_args, log=mock_logger)
+
+
+@pytest.fixture
+def mock_log_http_response(mocker):
+    mock_log_http_response = mocker.patch('datawave_cli.authorizations_interactions.log_http_response')
+    return mock_log_http_response
+
+
+@pytest.fixture
+def test_user():
+    return {
+        "proxiedUsers": [
+            {
+                "name": "test.user",
+                "auths": ["auth1", "auth2"],
+                "roles": ["AuthorizedUser"]
+            }
+        ]
+    }
+
+
+def test_whoami(mocker, authorization_interactions, mock_log_http_response, test_user):
+    """Tests the whoami function call to validate we are getting correct responses
+    and performing correct logic."""
+    mock_requests_get = mocker.patch('requests.get')
+    mock_response = mocker.Mock()
+    mock_response.status_code = 200
+    mock_response.json.return_value = test_user
+    mock_requests_get.return_value = mock_response

```

Comment:
Why aren't you using your new utils fixture?

Reply:
the function was written after...